plugin/evm/limitorders/service.go

```diff
@@ -101,3 +107,39 @@ func (api *OrderBookAPI) GetOpenOrders(ctx context.Context, trader string) OpenO
 
 	return OpenOrdersResponse{Orders: traderOrders}
 }
+
+// NewOrderBookData send a notification each time a new (header) block is appended to the chain.
+func (api *OrderBookAPI) NewOrderBookData(ctx context.Context) (*rpc.Subscription, error) {
+	notifier, supported := rpc.NotifierFromContext(ctx)
+	if !supported {
+		return &rpc.Subscription{}, rpc.ErrNotificationsUnsupported
+	}
+
+	rpcSub := notifier.CreateSubscription()
+
+	go func() {
+		var (
+			headers    = make(chan core.ChainHeadEvent)
+			headersSub event.Subscription
+		)
+
+		headersSub = api.backend.SubscribeChainHeadEvent(headers)
+		defer headersSub.Unsubscribe()
+
+		for {
+			select {
+			case <-headers:
+				orderBookData := api.GetDetailedOrderBookData(ctx)
+				notifier.Notify(rpcSub.ID, orderBookData)
```

Comment:
can pass orderBookData by reference here to avoid a copy

Reply:
done